programs/mango-v4/src/instructions/token_force_close_borrows_with_token.rs

```diff
@@ -0,0 +1,207 @@
+use anchor_lang::prelude::*;
+use fixed::types::I80F48;
+use std::cmp::min;
+
+use crate::accounts_ix::*;
+use crate::error::*;
+use crate::health::*;
+use crate::logs::TokenBalanceLog;
+use crate::state::*;
+
+// TODO: alternative ix name suggestions?
+pub fn token_force_close_borrows_with_token(
+    ctx: Context<TokenForceCloseBorrowsWithToken>,
+    asset_token_index: TokenIndex,
+    // token's mode is checked at #2
+    liab_token_index: TokenIndex,
+    max_liab_transfer: I80F48,
+) -> Result<()> {
+    let group_pk = &ctx.accounts.group.key();
+
+    require_neq!(asset_token_index, liab_token_index, MangoError::SomeError);
+    // TODO: should we enforce that asset token index is always USDC?
```

Comment:
Hmm, arguments could be made in both directions:
- We don't want force-closers to create random new borrows on users (use asset_token = BONK)
- We don't want to make USDC's role special

We could arbitrarily allow any token with liab weight <= 1.1? But that's also surprising. Hmm

Reply:
Any token with `asset_token_init_liab_weight <= liab_token_init_liab_weight + liab_token_liq_fee`? That way the liqee's health would be guaranteed to not decrease?

Reply:
makes sense, I'd use scaled weights or?